File: benchmarks/blur/java/MatrixUtil.java
package benchmarks.blur.java;

class MatrixUtil {
    /**
     * Flattens a 2D array into a 1D array.
     * 
     * @param arr 2D integer array
     * @return Flattened array
     */
    public static int[] flattenArray(int[][] arr) {
        if (arr.length == 0)
            throw new IllegalArgumentException("Array cannot be empty!");

        int height = arr.length;
        int width = arr[0].length;

        int[] flattened = new int[width * height];

        for (int y = 0; y < height; y++) {
            for (int x = 0; x < width; x++) {
                flattened[width * y + x] = arr[y][x];
            }
        }

        return flattened;
    }

    /**
     * Unflattens a 1D array into a 2D array with the given dimension.
     * 
     * @param arr    1D integer array to unflatten
     * @param width  Desired width of the 2D array
     * @param height Desired height of the 2D array
     * @return Unflattened array with the given dimensions
     * @throws IllegalArgumentException If the given dimensions do not match the
     *                                  array length
     */
    public static int[][] unflattenArray(int[] arr, int width, int height) {
        if (arr.length != width * height)
            throw new IllegalArgumentException("Given dimensions do not match the array!");

        int[][] unflattened = new int[height][width];

        for (int y = 0; y < height; y++) {
            for (int x = 0; x < width; x++) {
                unflattened[y][x] = arr[width * y + x];
            }
        }

        return unflattened;
    }

}

File: benchmarks/blur/python/matrix_util.py
class MatrixUtil:
    @staticmethod
    def flatten_array(arr: list[list[int]]) -> list[int]:
        """
        Flattens a 2D array into a 1D array.
        Args:
            arr: The 2D array to flatten.
        Returns:
            The flattened 1D array.
        """
        return [item for sublist in arr for item in sublist]

    @staticmethod
    def unflatten_array(arr: list[int], width: int, height: int) -> list[list[int]]:
        """
        Unflattens a 1D array into a 2D array with the given dimension
        Args:
            arr: 1D integer array to unflatten
            width: Desired width of the 2D array
            height: Desired height of the 2D array
        Returns:
            Unflattened array with the given dimensions
        Raises:
        Raises:
            ValueError: If the given dimensions do not match the array length. 
        """
        if len(arr) != width * height:
            raise ValueError("Given dimensions do not match the array!")

        unflattened: list[list[int]] = []

        for y in range(height):
            row = []
            for x in range(width):
                row.append(arr[width * y + x])
            unflattened.append(row)

        return unflattened
